### ISSUE TITLE
TCH migration

### ISSUE THREAD

**Issue by FL33TW00D** [huggingface/ratchet#191]
Our current test infrastructure involves raw strings: https://github.com/huggingface/ratchet/blob/master/crates/ratchet-core/src/ops/softmax.rs#L118

Whilst having the python code and the rust code directly next to each other is desirable, I think using tch might be better.

https://github.com/LaurentMazare/tch-rs

We still need the confidence of: "We match pytorch across 1000s of randomly generated tests", without the strings.
